src/store/reducers/transactions.js

```diff
@@ -56,38 +63,42 @@ const transactions = (state = initialState, action) => { // eslint-disable-line
     case actionTypes.updateTransactions:
       return {
         ...state, // Filter any newly confirmed transaction from pending
-        pending: state.pending.filter(pendingTransaction =>
-          action.data.confirmed.filter(transaction =>
-            transaction.id === pendingTransaction.id).length === 0),
+        pending: merge(state.pending, action.data.confirmed),
```

Comment:
`merge` is not what this filter was doing. It returns all transactions from `array1` that are not in `array2`, based on `id`. I cannot think of a good name for it though ☹️